src/twister2/specification_processor.py

```diff
@@ -222,13 +222,13 @@ def should_be_skip(
 def should_skip_for_toolchain(
     test_spec: YamlTestSpecification, platform: PlatformSpecification, used_toolchain_version: str
 ) -> bool:
-    if not platform.toolchain:
-        return False
-    if test_spec.toolchain_allow and not test_spec.toolchain_allow & set(platform.toolchain):
-        _log_test_skip(test_spec, platform, 'platform.toolchain not in testcase.toolchain_allow')
+    if test_spec.toolchain_allow and not (used_toolchain_version in test_spec.toolchain_allow):
```

Comment:
you can use: if test_spec.toolchain_allow and used_toolchain_version not in test_spec.toolchain_allow